### DISCUSSION TITLE
applying JWT plugin different option in same endponit

### DISCUSSION THREAD

**Discussion by liveforfun** [Kong/kubernetes-ingress-controller#2397] [Q&A] [answered]
hello I want to control the JWT token expiration time

https://docs.konghq.com/kubernetes-ingress-controller/1.3.x/guides/configure-acl-plugin/
https://docs.konghq.com/hub/kong-inc/jwt/

there is a maximum_expiration option in JWT
and I want some customers' keys to expire in some minutes and other customers' keys not to expire forever on the same endpoint 
is it possible??
the first time I thought I could set the multiple JWT in ingress or service but it's not working.
Does anybody think about that?
for instance, I want to do blow jobs done.

cumer1 key => PROXY_IP/api1 : expire 5 min
cumer2 key => PROXY_IP/api1 : no expire

**Comment by shaneutt** [comment, marked as answer]
There is no existing feature that would enable this. Plugins are not able to run more than once per request or vary the configuration they use based on request characteristics.

If you're the issuer of the JWT's you can set the `exp` yourself when issuing them, this would be the ideal solution.

If you're not the issuer and instead it's your client's are doing the issuing you will need to write a custom plugin to do this. The plugin could filter on each client's `Consumer` objects to check the expiration of the JWT in transit and instigate custom expiration there.
